### PR TITLE
Address issues #568, #561

### PR DESCRIPTION
**Pull request by awav**
* Improves initialization checking mechanizm
* Iterator still cannot be checked if it were initialized for particular session or not, hence it will just always be initialized.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ install:
   - python setup.py install
 
 script:
-  - pytest -x --cov=./ -d --tx 3*popen//python=python3.6 --pyargs tests
```
Comment (markvdw) [inline .travis.yml]: I would rather that the tests fail on the server asap. One can then run the tests locally for full coverage.

But I don't have a strong opinion.

**Comment by markvdw** [comment]
This looks great. I'll review. We should try to get this in asap.

**Comment by markvdw** [comment]
I'm happy to merge these changes as soon as the test coverage is what we require.

File: .travis.yml (hunk 1 above)
Comment (awav) [reply .travis.yml]: Disagree, travis and developer's environments are different and can produce unlike results, which *must* not happen, but still it is important to see whole report and all errors.

**Comment by awav** [comment]
@markvdw you may want to squash, but in fact github show most of all of these commits for no reason, probably someone created a branch from mine when it hadn't merged into master yet, but squashed later (it is one of the drawbacks of squashing). This branch has only 4 commits, a couple of them have references to issues in github and were bounded.  I propose do not squash this branch.

**Comment by markvdw** [comment]
Also, what does it mean for an "iterator to always be initialised"? Can this cause variables to be reset, like in earlier versions of the code?

**Comment by awav** [comment]
@markvdw, This is a plug for now, until we find another proper solution. Everytime whenever you call autoflow, iterator will be re-initialized. It is a problem of TensorFlow - iterator doesn't provide an interface for checking if it were initialized for a session or not.

**Pull request merged**